js/background.js

```diff
@@ -100,9 +100,21 @@
         return new textsecure.SyncRequest(textsecure.messaging, messageReceiver);
     };
 
+    Whisper.events.on('shutdown', function() {
+      if (messageReceiver) {
+        messageReceiver.close().then(function() {
+          messageReceiver = null;
```

Comment:
There's only one place in the code that starts up the `MessageReceiver` and that's in `init()` in `background.js`, which checks `Whisper.Migration.inProgress()` first.